baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/IllegalSafeLoggingArgument.java

```diff
@@ -144,12 +160,42 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
         return Description.NO_MATCH;
     }
 
-    private static SuggestedFix getSuggestedFix(MethodInvocationTree tree, VisitorState state, Safety argumentSafety) {
+    private void handleTypeArguments(ExpressionTree tree, VisitorState state) {
+        Type type = ASTHelpers.getResultType(tree);
+        if (type != null && !type.getTypeArguments().isEmpty()) {
+            List<Type> resultTypeArguments = type.getTypeArguments();
+            List<TypeVariableSymbol> parameterTypes = type.tsym.getTypeParameters();
+            if (parameterTypes.size() == resultTypeArguments.size()) {
```

Comment:
Do you have an example of where these sizes would not be equal?

Reply:
I do not, but I’ve run into cases where generic type info is missing due to compiler optimizations, so I figured it was best not to risk it.